### PR TITLE
Support ACK Timeout event on radio layer.

### PR DESCRIPTION
**Pull request by xiaom-GitHub**

### CODE REVIEW

File: examples/platforms/efr32/radio.c [hunk 1]
```diff
@@ -770,15 +773,19 @@ void efr32RadioProcess(otInstance *aInstance)
     CORE_EXIT_CRITICAL();
 }
 
-void RAILCb_RfReady(void)
+void RAILCb_RxAckTimeout(void)
 {
+    if (sIsAckRequested)
```
Comment (jwhui) [inline examples/platforms/efr32/radio.c]: Is this check necessary?  If so, should we set `sIsAckRequested` to `false` here?

Comment (xiaom-GitHub) [inline examples/platforms/efr32/radio.c]: This check is added only for unicast packet which needs ACK from peer. `sIsAckRequested` will be set to `false` if no ACK requested (for example, transmit broadcast packet) in line 322. If ACK is received, at line 766, `sIsAckRequested` will be set to  `false`.

Comment (jwhui) [inline examples/platforms/efr32/radio.c]: But will this callback be called when no ack is requested?

Comment (xiaom-GitHub) [inline examples/platforms/efr32/radio.c]: Yes, observing from testing between two dev boards (add log info at this callback), notice that for broadcast packet (e.g. advertisement), the log will be printed as well. so I add one check here only for unicast packet.

Comment (jwhui) [inline examples/platforms/efr32/radio.c]: Seems odd to be waiting for an ACK when the packet does not request one.  How about using the following sequence when no ack is requested?

```
    RAIL_TxOptions_t txOption;
    txOption.waitForAck = false;
    RAIL_Status_t status = RAIL_TxStartWithOptions(0, &txOption, NULL, NULL);
```

Comment (xiaom-GitHub) [inline examples/platforms/efr32/radio.c]: yes, will update it soon. Thanks.

**Pull request merged**